api/envoy/api/v2/route/route.proto

```diff
@@ -437,6 +439,22 @@ message RouteAction {
     google.protobuf.Duration per_try_timeout = 3 [(gogoproto.stdduration) = true];
   }
 
+  // Specifies the idle timeout for the route. If not specified, there is no
+  // stream level timeout. This is distinct to :ref:`timeout
```

Comment:
I'm OK with setting something here like 30s. I think we should make it > default response timeout (15s), to reduce the chance of flakes in timeout due to idle vs. due to total response timeout. Does this work?

Reply:
Yeah I think a default here is fine as long as it's greater than the default of 15s. However, I don't think that's the whole story. For people that have increased the timeout above 15s (or disabled it), this could cause problems. So, I think we either have to not have a default here, or actually have the logic to make the default if not set at least as long as the configured request timeout. Thoughts?

Reply:
There's also the per-connection idle timeout to consider as well. I think if we want this to be unsurprising to existing users we will have to make it a max function of all the relevant timeouts. Might be easier to leave it to default off.

Reply:
TBH that's my feeling also. I would leave it without default.

Reply:
I think it's fundamentally incorrect to have the default behavior of an HTTP proxy never time out POST requests.  What do we think of adding the feature as default-off, and having a TODO to add a default timeout later as "deprecating permanent lived POSTS" using our standard multi-release-cycle for deprecated features? That'd give folks with different timeouts plenty of time to adjust them but make sure the long term behavior of Envoy would be to not leak connections if they time out on the request path?

Reply:
I'm less worried about the trickle attack DoS vector (Envoy has plenty of those yet) and more about folks losing their connection without the FIN being transmitted.  This is a real thing that happens frequently on internet facing traffic and will cause connection (so fd and memory) leakage without the buffer filter present.  It's true that either having a default request timeout would address my concern as well, I was just hoping to fix the resource leakage sooner rather than later.
If you really think this should be off by default and the default timeout should be in the should-be-moved request timeout I'll cede the issue.

Reply:
I agree this is a big problem, and I'm kicking myself for ending up where we are, but what's done is done. The issue mainly is that Lyft solves this problem w/ the buffer filter at the edge as we discussed offline. Totally agree we need a built-in timeout, but I'm also concerned that setting a default here is going to break people. I think we should have some big warning in the docs somewhere about this (where?) and I guess we could do a timeout here but IMO the default needs to be really high if we set one. E.g., like 5 minutes or something. Or we need to do some complex logic to infer the timeout somehow from the other timeouts people set.

Reply:
I think a default idle-timeout of 5 minutes (or longer) should be fine. I also agree that having a POST overall timeout is useful, but I think that one is impossible to have a reasonable default value for.

Reply:
OK, 5 minutes default, going once, twice, ...

Reply:
Works for me!  As said I'm also fine landing and then changing the default in a separate PR (even a after a long window) if you want to have the risky change minimal.